### DISCUSSION TITLE
Should kombu support consuming messages which are published by other systems?

### DISCUSSION THREAD

**Discussion by ZLotusRain** [celery/kombu#2326] [Q&A]
As far as i know,messages received by kombu must have specific properties, but in some situation we want to use kombu to consume messages which don't have headers and properties and other attributes, and yes we can consider the message as message.body in kombu.

So should kombu support consuming messages which are published by other systems or in other words don't have specific attributes

**Comment by TayfurYldz** [comment]
Yes. Kombu can consume messages produced by non-Kombu systems; the producer does not have to be Celery/Kombu. The interoperability boundary is the AMQP message's body plus content metadata, not the identity of the publishing library.

What matters is how you ask Kombu to decode the body. Kombu's Consumer uses content_type/content_encoding (and its serializer registry / accept list) to deserialize a message. A foreign publisher that omits those properties cannot expect Kombu to infer "this byte sequence is JSON" reliably.

If the foreign messages are deliberately just opaque bytes, consume them as raw content instead of pretending they use Kombu's JSON serializer. You can register/accept the appropriate content type when the other producer sets one, or handle the raw body and decode it in your callback according to your own protocol.

For cross-system JSON, the clean contract is for the foreign publisher to set at least:

content_type = application/json
content_encoding = utf-8

and publish valid JSON bytes. Then configure the consumer's accept list accordingly.

Headers are application-specific and can be absent unless your application/Celery protocol requires them. Celery tasks are a different case: a Celery worker expects the Celery task message protocol, so an arbitrary AMQP message is not automatically a valid Celery task merely because Kombu can receive it.

So: Kombu transport interoperability is supported; Celery-task interoperability requires Celery's message schema; and serializer metadata should be explicit when you want automatic deserialization.
